### PR TITLE
Copy Wordpress authors and excerpt to posts, use in backporting

### PR DESCRIPTION
**Pull request by danyx23**
This PR collects the authors from wordpress (all of them, with their order) and the excerpt (what is shown as the subtitle in both old and new style articles). The posts table in our mysql database is extended with two columns, the scripts to bulk fetch wordpress posts and the hook to update single posts on wordpress save are updated, and the information is used in backporting to create gdoc articles. This is part of issue #1870 

When rolling this out to prod:
- [ ] Run the /db/syncPostsToGrapher script on prod to fill the new columns
- [ ] Run the /db/migrateWpPostsToArchieMl script on prod to copy this information into the archieml json

### CODE REVIEW

**Review by ikesau — APPROVED** [review]
Nice work on that query!

Ran it against my local WP and checked the bylines on a handful of the corresponding posts and everything matched.

Haven't tested the full flow of the PR, let me know if you'd like me to.

File: baker/postUpdatedHook.ts [hunk 1]
```diff
@@ -17,12 +17,24 @@ const zeroDateString = "0000-00-00 00:00:00"
...
-        .getSingleton()
-        .query(
-            "SELECT * FROM wp_posts WHERE ID = ? AND post_status != 'trash'",
-            [postId]
-        )
+    const rows = await wpdb.getSingleton().query(
+        `--sql
+        -- we select all the fields from wp_posts and then we also
+        -- json array aggregate the authors. The authors come as strings like
+        -- Charlie Giattino Charlie Giattino Charlie Giattino 44 charlie@ourworldindata.org
```
Comment (marcelgerber) [inline baker/postUpdatedHook.ts:24]: how did that happen??

File: baker/postUpdatedHook.ts [hunk 2]
```diff
@@ -17,12 +17,24 @@ const zeroDateString = "0000-00-00 00:00:00"
...
+        -- so we use regexp_replace to cut out the first two words
+        SELECT p.*, JSON_ARRAYAGG(
+            JSON_OBJECT('author',
+                regexp_replace(t.description, '^([[:alnum:]]+) ([[:alnum:]]+) .+$' , '$1 $2'),
```
Comment (marcelgerber) [inline baker/postUpdatedHook.ts:28]: Okay, I hate to break this to you, but: This doesn't work for `Esteban Ortiz-Ospina` (see https://regexr.com/782mi).
It also won't work as intended for double names.

File: db/migration/1675447441269-AddAuthorsToPosts.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class AddAuthorsToPosts1675447441269 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE posts
+        ADD COLUMN authors JSON comment 'json array of objects with field author in the form "firstname lastname" and field order'
```
Comment (marcelgerber) [inline db/migration/1675447441269-AddAuthorsToPosts.ts:6]: TIL that you can add comments to SQL columns! who knew.

**Review by marcelgerber — CHANGES_REQUESTED** [review]

File: baker/postUpdatedHook.ts (hunk 1 above)
Comment (danyx23) [reply baker/postUpdatedHook.ts:24]: It's a plugin that does this. Don't ask :)

File: baker/postUpdatedHook.ts (hunk 2 above)
Comment (danyx23) [reply baker/postUpdatedHook.ts:28]: Oh and to clarify - this regex is still not perfect and as you say, double names without dashes are still a problem. But I compared this with the authors we actually have so far and this seems to capture all of the correctly. The alternative would be to split the tokens at whitespaces and work out the structure but this one seems more fitting for this particular case

**Comment by danyx23** [comment]
Nice job nerd-sniping me into writing a `partition by` query to figure out the created_at_in_wordpress data, @ikesau 😅 . Can you have another quick look? I liked the idea and added a new column to the posts table that is now filled in the sync script and the wordpress hook

File: baker/postUpdatedHook.ts [hunk 4]
```diff
@@ -19,21 +19,58 @@ const syncPostToGrapher = async (
...
+        ),
+        -- and then we use another CTE to find the first revision
+        first_revision as (
+            select
+                post_date as created_at,
+                post_parent as post_id
+            from
+                revisions
+            where
+                row_num = 1),
```
Comment (marcelgerber) [inline baker/postUpdatedHook.ts:47]: would that not be easier with the `LIMIT 1`, as ike had initially suggested?

Comment (danyx23) [reply baker/postUpdatedHook.ts]: Ah nice, in this one that is definitely the better choice. I wrote the other one first and then this was the simple adaptation but you're right, better to make it simpler. Thanks!

**Review by ikesau — APPROVED** [review]
Tested the `created_at` logic locally on a couple of articles. LGTM!

**Pull request merged**